server/server.js

```diff
@@ -1,15 +1,129 @@
+require("dotenv").config();
+
 const express = require("express");
+const cors = require("cors");
+const morgan = require("morgan");
+
 const app = express();
-const port = process.env.PORT || 5000;
 
-app.listen(port, () => console.log(`Listening on port ${port}`));
+const { Pool } = require("pg");
+
+const db = new Pool({
+  user: process.env.DB_USERNAME,
+  host: process.env.DB_HOST,
+  database: process.env.DB_DATABASE,
+  password: process.env.DB_PASSWORD,
+  port: process.env.DB_PORT,
+  // ssl: {
+  //   rejectUnauthorized: false,
+  // },
+});
+// console.log(db);
+
+app.use(cors());
+app.use(morgan("dev"));
+
+app.use(express.static("public"));
+app.use(express.json());
+app.use(express.urlencoded({ extended: true }));
 
-// Store and retrieve your videos from here
-// If you want, you can copy "exampleresponse.json" into here to have some data to work with
-let videos = [];
+//$ npm run dev
+
+// app.use((req, res, next) => {
+//   res.header("Access-Control-Allow-Origin", "*"); // Allow requests from any domain
+//   next();
+// });
 
-// GET "/"
 app.get("/", (req, res) => {
-  // Delete this line after you've confirmed your server is running
   res.send({ express: "Your Backend Service is Running" });
 });
+
+app.get("/video", async (req, res) => {
+  try {
+    const order = req.query.order === "asc" ? "ASC" : "DESC";
+    const query = `SELECT * FROM videos ORDER BY rating ${order}`;
+    const request = await db.query(query);
+    const response = request.rows;
+    // console.log(response);
+    res.status(200).json(response);
+  } catch (err) {
+    console.error(err);
+    res.status(500).json({ message: "Internal server error" });
+  }
+});
+
+app.post("/video", async (req, res) => {
+  const { title, url } = req.body;
+  // console.log("post video req.body", req.body);
+  const id = url.split("embed/")[1];
```

Comment:
What assumptions are you making about input from users here? What would happen if those violations were violated? How could they be?

Reply:
Regex changes it before the request is sent, so it prevents that issue on the front-end side, passing the clean URL to the server.